lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -42,7 +42,196 @@ const { makePathsRelative, parseResource } = require("../util/identifier");
 /** @typedef {import("../util/smartGrouping").GroupConfig<any, object>} GroupConfig */
 /** @typedef {import("./StatsFactory")} StatsFactory */
 
-/** @typedef {Asset & { type: string, related: ExtendedAsset[] }} ExtendedAsset */
+/**
+ * @typedef {Object} StatsOutput
+ * @property {any=} env
+ * @property {string=} name
+ * @property {string=} hash
+ * @property {string=} version
+ * @property {number=} time
+ * @property {number=} builtAt
+ * @property {boolean=} needAdditionalPass
+ * @property {string=} publicPath
+ * @property {string=} outputPath
+ * @property {Record<string, string[]>=} assetsByChunkName
+ * @property {StatsAsset[]=} assets
+ * @property {number=} filteredAssets
+ * @property {StatsChunk[]=} chunks
+ * @property {StatsModule[]=} modules
+ * @property {number=} filteredModules
+ * @property {Record<string, StatsChunkGroup>=} entrypoints
+ * @property {Record<string, StatsChunkGroup>=} namedChunkGroups
+ * @property {StatsError[]=} errors
+ * @property {number=} errorsCount
+ * @property {StatsError[]=} warnings
+ * @property {number=} warningsCount
+ * @property {StatsOutput[]=} children
+ * @property {any=} logging
+ */
+
+/**
+ * @typedef {Partial<Asset> & {
```

Comment:
Set this as `Partial` because the output does not contain the `source` property which means when we cast the output, it will fail as it is expecting `source` to be defined.